patronload/patron.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import re
+from datetime import date
+from typing import Any, Generator
+
+from bs4 import BeautifulSoup
+from dateutil.relativedelta import relativedelta
+
+logger = logging.getLogger(__name__)
+
+
+def create_patron_dicts(
+    fields: list[str], patron_records: list[tuple]
+) -> Generator[dict, None, None]:
+    """
+    Create dicts of patron data from a list of patron records.
+
+    Patron records are retrieved as tuples from the Data Warehouse and
+    the dicts are returned via a generator.
+
+    Args:
+        fields: The fields that serve as keys in the dict.
+        patron_records: A list of patron record tuples.
+    """
+    for patron_record in patron_records:
+        patron_dict = {}
+        for field in fields:
+            patron_dict[field] = patron_record[fields.index(field)]
+        yield patron_dict
+
+
+def format_phone_number(phone_number: str) -> str:
+    """
+    Format a string of 10 numbers as a phone number with dashes.
+
+    Args:
+        phone_number: A string that may contain a phone number.
+    """
+    return re.sub(r"(\d{3})(\d{3})(\d{4})", r"\1-\2-\3", phone_number)
+
+
+def populate_patron_common_fields(
+    patron: BeautifulSoup, patron_dict: dict[str, Any]
+) -> BeautifulSoup:
+    """
+    Populate the fields common to both staff and student patron records.
+
+    Args:
+        patron: An XML template for a patron record
+        patron_dict: A dict of patron data used to populate an XML template.
+    """
+    six_months = date.today() + relativedelta(months=+6)
+    two_years = six_months + relativedelta(years=+2)
+
+    patron.find_all("primary_id")[0].string = (
+        f"{patron_dict['KRB_NAME_UPPERCASE']}@MIT.EDU"
+        if patron_dict["KRB_NAME_UPPERCASE"]
+        else patron_dict["EMAIL_ADDRESS"]
+    )
+    patron.find_all("expiry_date")[0].string = six_months.strftime("%Y-%m-%d") + "Z"
```

Comment:
These date strings are the same for every record, so I would set them elsewhere rather than recreating them every time you process a record.

Reply:
Shifted it in the new commit, let me know if you think it should be elsewhere